cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/SystemArguments.java

```diff
@@ -295,6 +298,20 @@ public static RetryStrategy getRetryStrategy(Map<String, String> args, ProgramTy
     return RetryStrategies.fromConfiguration(policyConf, keyPrefix);
   }
 
+  /**
+   *
+   * @param args the arguments to lookup instrumentation settings
+   * @param defaultProcessTimingEnabled default value to use if instrumentation settings are missing from the arguments
+   */
+  public static boolean isProcessTimingEnabled(Map<String, String> args,
+                                               @Nullable boolean defaultProcessTimingEnabled) {
+    Boolean value = getBoolean(args, INSTRUMENTATION_KEY);
+    if (value != null) {
+      return value.booleanValue();
+    }
+    return defaultProcessTimingEnabled;
```

Comment:
I think you don't really need this `defaultProcessTimingEnabled`, we can add the check in `SmartWorkflow.java`, also not sure if that should be called default value?